sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -380,6 +380,11 @@ class SQLContext(@transient val sparkContext: SparkContext)
     jsonRDD(json.rdd, schema)
   }
 
+  @Experimental
+  def inferJsonSchema(json: String): StructType = {
```

Comment:
Thanks for pointing this out! I agree we should provide a utility for the general purpose, let me know if I can offer help, or should I close this PR?